components/fallback/validate.hpp

```diff
@@ -66,6 +69,33 @@ namespace Files {
         for (std::vector<std::string>::const_iterator it = tokens.begin(); it != tokens.end(); ++it)
             eSV->mVector.push_back(EscapeHashString(*it));
     }
+
+    struct EscapePath {
+        boost::filesystem::path mPath;
+
+        static PathContainer toPathContainer(const std::vector<EscapePath> & escapePathContainer);
+    };
+
+    typedef std::vector<EscapePath> EscapePathContainer;
+
+    PathContainer EscapePath::toPathContainer(const EscapePathContainer & escapePathContainer)
+    {
+        PathContainer temp;
+        for (EscapePathContainer::const_iterator it = escapePathContainer.begin(); it != escapePathContainer.end(); ++it)
+            temp.push_back(it->mPath);
+        return temp;
+    }
+
+    std::istream & operator>> (std::istream & istream, EscapePath & escapePath)
+    {
+        boost::iostreams::filtering_istream filteredStream;
+        filteredStream.push(unescape_hash_filter());
```

Comment:
This is where we add a filter to restore the contents of the input stream to its former state (as it's now made it through boost's comment ignoring system safely). However, this causes an "unresolved external symbol" error. Using an ```escape_hash_filter``` instead works fine, even though both filters are (as far as I understand) declared and defined in the same place.